src/DataShare/DataShare/ProviderShareSubscriptions/GrantAzDataShareSubscriptionAccess.cs

```diff
@@ -80,8 +80,7 @@ public class GrantAzDataShareSubscriptionAccess : AzureDataShareCmdletBase
             Mandatory = true,
             ValueFromPipelineByPropertyName = true,
             HelpMessage = "The share subscription id of the provider share subscription",
-            ParameterSetName = ParameterSetNames.ResourceIdParameterSet)]
-        [ResourceIdCompleter(ResourceTypes.ShareSubscription)]
+            ParameterSetName = ParameterSetNames.ProviderShareSubscriptionParameterSet)]
         [ValidateNotNullOrEmpty]
         public string ShareSubscriptionId { get; set; }
```

Comment:
Call this byProviderShareSubscriptionId because we would have to add implementation with object too